python/packages/autogen-core/src/autogen_core/model_context/_token_based_chat_completion_context.py

```diff
@@ -0,0 +1,72 @@
+from typing import List, Sequence
+from autogen_core.tools import Tool, ToolSchema
+
+from pydantic import BaseModel
+from typing_extensions import Self
+import tiktoken
+
+from .._component_config import Component
+from ..models import FunctionExecutionResultMessage, LLMMessage
+from ._chat_completion_context import ChatCompletionContext
+
+
+class TokenBasedChatCompletionContextConfig(BaseModel):
+    token_limit: int
+    model: str
+    initial_messages: List[LLMMessage] | None = None
+
+
+class TokenBasedChatCompletionContext(ChatCompletionContext, Component[TokenBasedChatCompletionContextConfig]):
```

Comment:
I think we can name it "TokenLimitedChatCompletionContext"

Reply:
sure just updated